src/Main.elm

```diff
@@ -632,17 +671,130 @@ modalView : Model -> Html Msg
                     ]
                 ]
             ]
         ]
 
 
+testDisplay : Maybe Menu -> Html Msg
+testDisplay menu =
+    label [ style "display" "flex", style "align-items" "center" ]
+        [ input
+            [ type_ "radio"
+            , name "value"
+            , onClick (OpenMenu (CustomDifficultyMenu 50 50 50))
+            , checked True
+            , style "margin" "4px 8px"
+            ]
+            []
+        , text "Custom Difficulty"
+        ]
+
+
+radiobutton : String -> Difficulty -> Difficulty -> Html Msg
+radiobutton settingLabel difficulty currentGameDifficulty =
+    label [ style "display" "flex", style "align-items" "center" ]
+        [ input
+            [ type_ "radio"
+            , name "value"
+            , onClick (SetDifficulty difficulty)
+            , checked (difficulty == currentGameDifficulty)
+            , style "margin" "4px 8px"
+            ]
+            []
+        , text settingLabel
+        ]
+
+
+customRadioButton : Maybe Menu -> Html Msg
+customRadioButton currentMenu =
+    let
+        isCustomDifficultyMenu menu2 =
+            case menu2 of
+                CustomDifficultyMenu _ _ _ ->
+                    True
+
+                _ ->
+                    False
+    in
+    case currentMenu of
+        Nothing ->
+            div [] []
+
+        Just menu ->
+            case menu of
+                DifficultyMenu ->
+                    label [ style "display" "flex", style "align-items" "center" ]
+                        [ input
+                            [ type_ "radio"
+                            , name "value"
+                            , onClick (OpenMenu (CustomDifficultyMenu 20 20 50))
+                            , checked (isCustomDifficultyMenu menu)
+                            , style "margin" "4px 8px"
+                            ]
+                            []
+                        , text "Custom Difficulty"
+                        ]
+
+                CustomDifficultyMenu x y z ->
+                    let
+                        handleInput inputString =
+                            case String.toInt inputString of
+                                Nothing ->
+                                    0
+
+                                Just int ->
+                                    int
+                    in
+                    div []
+                        [ input
+                            [ type_ "radio"
+                            , name "value"
+                            , onClick (OpenMenu (CustomDifficultyMenu x y z))
+                            , checked (isCustomDifficultyMenu menu)
+                            , style "margin" "4px 8px"
+                            ]
+                            []
+                        , text "Custom Difficulty"
+                        , br [] []
+                        , text "Width"
+                        , input
+                            [ type_ "text"
```

Comment:
you could do `type_ "number"`, so the browser does some of the work validating the input.